sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -259,8 +259,23 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
    * @throws IllegalArgumentException if a field with the given name does not exist
    */
   def apply(name: String): StructField = {
-    nameToField.getOrElse(name,
-      throw new IllegalArgumentException(s"""Field "$name" does not exist."""))
+    if (name.contains('.')) {
```

Comment:
IIUU, this will drop the support to access the field name containing `.` (e.g. `"a.b"`) which can be accessed via `` "`a.b`" ``.